app/models/user.rb

```diff
@@ -6,9 +6,24 @@ def self.from_omniauth(auth)
   end
 
   def assign_from_auth(auth)
-    auth_attrs = auth.info
-    self.email = auth_attrs.email
-    self.name = auth_attrs.name
     self.uid = auth.uid
+
+    assign_attrs(auth.info)
+    extra_attributes_from_auth(auth) if auth.extra
+  end
+
+  private
+
+  def assign_attrs(auth_attrs)
+    self.email = auth_attrs.email
+    self.first_name = auth_attrs.first_name
+    self.middle_name = auth_attrs.middle_name
+    self.last_name = auth_attrs.last_name
+  end
+
+  def extra_attributes_from_auth(auth)
+    extra_attrs = auth.extra.raw_info
+    self.social_security_number = extra_attrs[:ssn]
+    self.phone = extra_attrs[:mobile]
   end
```

Comment:
The Why is to separate the responsibilities of the app's classes and objects, and to make testing easier and faster. Ideally, models should only deal with persistence and associations, and sometimes validations. We want to avoid making them "fat" by adding business logic to them. 

Here are some relevant posts and talks: 
http://jamesgolick.com/2010/3/14/crazy-heretical-and-awesome-the-way-i-write-rails-apps.html
http://blog.codeclimate.com/blog/2012/10/17/7-ways-to-decompose-fat-activerecord-models/
http://naildrivin5.com/blog/2014/05/27/rails-does-not-define-your-application-architecture.html
https://www.netguru.co/blog/service-objects-in-rails-will-help
https://www.youtube.com/watch?v=WpkDN78P884

Coupling business logic with persistence objects (User model for example) usually means we have to interact with the database in order to test the business logic. Invoking the database slows down tests. Ideally, tests should never touch the database. By using a separate class for business logic, we encapsulate that logic, decouple it from the User model, and make it easier and faster to unit test.

An example of putting this logic in a separate class is the [OmniauthAuthorizer](https://github.com/18F/identity-idp/blob/master/app/services/omniauth_authorizer.rb) in the idp repo. Note that testing that class still requires creating users in the database, and that's something I'm working on to fix.